### DISCUSSION TITLE
Unexpected Build Size for ONNX Runtime Mobile

### DISCUSSION THREAD

**Discussion by michaelboth** [microsoft/onnxruntime#6551] [Other Q&A] [answered]
Hi,
I read the Microsoft article on ONNX Runtime Mobile ([link](https://cloudblogs.microsoft.com/opensource/2020/10/12/introducing-onnx-runtime-mobile-reduced-size-high-performance-package-edge-devices/)).

It stated the the full build size of ONNX Runtime is about 7.5 MB, and that an ONNX Runtime Mobile minimal build is significantly less. In my experiments, I'm getting orders of magnitude larger if referring to the `libonnxruntime.so` file. Here's what I'm getting:

Ubuntu 18.04 Linux (full build): 522,457,648 bytes
Android (minimal for arm64-v8a): 92,731,008 bytes
Android (minimal for arm64-v8a plus `--android_cpp_shared`): 87,751,856 bytes

All three variations seem to be 100x larger than expected according to the above article.

The `required_operators.config` file (needed for the minimal build, using `--optimization_level basic`) contains:
```
# Generated from model/s in .
ai.onnx;1;Conv,GlobalAveragePool
ai.onnx;6;LeakyRelu
ai.onnx;8;MaxPool
ai.onnx;9;Flatten
```
Here's the build commands for the above builds:
Linux (full build):
`./build.sh --use_cuda --cudnn_home /usr/local/cuda/lib64 --cuda_home /usr/local/cuda --build_shared_lib`

Android (minimal for arm64-v8a):
`./build.sh --config RelWithDebInfo --android --android_sdk_path /home/michael/Android --android_ndk_path /home/michael/Android/Sdk/ndk/22.0.7026061 --android_abi arm64-v8a --android_api 29 --minimal_build extended --use_nnapi --disable_ml_ops --disable_exceptions --build_shared_lib --skip_tests --include_ops_by_config ../inference_test/required_operators.config`

Android (minimal for arm64-v8a plus `--android_cpp_shared`):
`./build.sh --config RelWithDebInfo --android --android_sdk_path /home/michael/Android --android_ndk_path /home/michael/Android/Sdk/ndk/22.0.7026061 --android_abi arm64-v8a --android_api 29 --minimal_build extended --use_nnapi --disable_ml_ops --disable_exceptions --build_shared_lib --android_cpp_shared --skip_tests --include_ops_by_config ../inference_test/required_operators.config`

Am I building incorrectly, looking at the wrong .so file, or was it a misprint in the above article?
I'm trying to avoid an oversided Android app.
Thanks for any guuidance.
Michael

**Comment by manashgoswami** [comment, marked as answer]
For the mobile build, the config is `MinSizeRel`. Can you try with `--config=MinSizeRel` in the build command? 

The build instructions are [here](https://github.com/microsoft/onnxruntime/blob/master/docs/ONNX_Runtime_for_Mobile_Platforms.md#2-create-the-minimal-build).

**Reply by michaelboth** [reply]
Thanks, and great catch. `MinSizeRel` significantly reduced the size. What is big difference from `RelWithDebInfo`.

**Reply by skottmckay** [reply]
The biggest difference is that when you build with MinSizeRel we automatically strip the debug symbols from the binary. It is possible to manually do that by running `strip` on the libonnxruntime.so produced by a RelWithDebInfo build. The `strip` binary is available in the Android NDK.
